openassessment/xblock/submission_mixin.py

```diff
@@ -552,19 +575,28 @@ def submission_path_and_context(self):
 
             file_urls = []
 
-            for i in range(self.MAX_FILES_COUNT):
-                file_url = self._get_download_url(i)
-                file_description = ''
-                file_name = ''
-                if file_url:
-                    try:
-                        file_description = saved_files_descriptions[i]
-                        file_name = saved_files_names[i]
-                    except IndexError:
-                        pass
-                    file_urls.append((file_url, file_description, file_name))
-                else:
-                    break
+            if saved_files_descriptions == []:
+                #This is the old behavior, required for a tiny corner case and should eventually be removed
+                for i in range(self.MAX_FILES_COUNT):
+                    file_url = self._get_download_url(i)
+                    file_description = ''
+                    file_name = ''
+                    if file_url:
+                        try:
+                            file_description = saved_files_descriptions[i]
+                            file_name = saved_files_names[i]
+                        except IndexError:
+                            pass
+                        file_urls.append((file_url, file_description, file_name))
+                    else:
+                        break
+            else:
+                for i, (file_description, file_name) in enumerate(zip(saved_files_descriptions, saved_files_names)):
+                    if file_description is None:
+                        file_urls.append((None, None, None))
```

Comment:
Oh, wait, yeah, we have to pass it to the front-end so that we can later use the correct index if we want to delete the file, right?  If that's the case, would you mind adding a comment here to explain that necessity?